src/openms/source/SIMULATION/LABELING/BaseLabeler.cpp

```diff
@@ -206,29 +207,29 @@ namespace OpenMS
         // adduct compositions we use the adduct-string as indicator to find the groups
         Map<String, std::set<FeatureHandle, FeatureHandle::IndexLess> > charge_mapping;
 
-        for (ConsensusFeature::iterator cf_iter = (*cm_iter).begin(); cf_iter != (*cm_iter).end(); ++cf_iter)
+        for (const FeatureHandle& cf : (cm))
         {
-          IntList feature_indices = id_map[String((*cf_iter).getUniqueId())];
+          IntList feature_indices = id_map[String((cf).getUniqueId())];
 
-          for (IntList::iterator it = feature_indices.begin(); it != feature_indices.end(); ++it)
+          for (int& feat : feature_indices)
```

Comment:
for build in types like int I would use a copy without the reference: 
for(const int f_index : feature_indices)
might even be faster